Custom header (title, subtitle, and logo)

hav142 opened the pull request:
This fixes issue #725.

How to test the feature:
- Add new config to prism.json for each country, i.e:
```
{
  "country": "Mongolia",
  "header": {
     "title": "PRISM v.3",
     "subtitle": "test subtitle",
     "logo": "images/wfp_logo.png"
   },
  "icons": {
...
```
- The logo will dynamically adjust its size based on the height, allowing it to accommodate logos of different size.

before (or without config):
![image](https://github.com/WFP-VAM/prism-app/assets/25983260/f539fea8-6af1-4413-9cab-6c0e8dd4773c)

after:
![image](https://github.com/WFP-VAM/prism-app/assets/25983260/81b6de94-b6aa-4554-9f43-1d50d8296af8)
![image](https://github.com/WFP-VAM/prism-app/assets/25983260/6f0bba18-51da-4ec2-8fb8-363b46346b5a)

HarryMytilinaios commented:
Sorry to intervene @hav142 @ericboucher @wadhwamatic in this PR I know I am not part of the reviewers, but because this PR I think it has similarities with an issue assigned to me https://github.com/WFP-VAM/prism-app/issues/629 . I'd like to ask on how are we supposed to translate those keys (if we wanted to, of course), that are part of the `prism.json`. Because as far as I know, the translations are derived from the `tranlsations.json` file of every country.

ericboucher commented on frontend/src/components/NavBar/index.tsx:
```diff
@@ -48,19 +51,31 @@ function NavBar({ classes }: NavBarProps) {
     </Grid>
   ));
 
+  const title = get(appConfig, 'title', 'PRISM');
```
I think you should create a `header` config object within the config instead of having everything at the top level

hav142 replied:
Thanks @ericboucher  for the correction. So, all country will have default config where the title is set to 'PRISM' like `'header': { title: 'PRISM' }`?

ericboucher replied:
So we would do something like that:

{title, subtitle, logo} = get(appConfig, 'header', {title: 'PRISM'});

ericboucher commented on frontend/src/components/NavBar/index.tsx:
```diff
@@ -48,19 +51,31 @@ function NavBar({ classes }: NavBarProps) {
+                component={Link}
+                to="/"
+              >
+                {t(title)}
```
@HarryMytilinaios the English version can be passed as the key directly as it is done here

wadhwamatic commented:
This is a good start @hav142, thanks! When I add an empty 'title' to the config, we just see the logo, which is good. I tried extending over muliple lines and clearly there's a limit which is fine as I don't see a need to have multiline titles / subtitles. However, please do look at https://github.com/WFP-VAM/prism-app/pull/847 and see how this will work adaptively 

And as @HarryMytilinaios mentioned, let's make sure we have a means to handle translations. I didn't try that yet but can once you've confirmed that is also ready for testing.

<img width="1664" alt="Screenshot 2023-05-31 at 00 02 02" src="https://github.com/WFP-VAM/prism-app/assets/3343536/6275ca6d-0a1a-4eae-bc9a-f74ccdcdf59f">

after reducing the page width:
<img width="966" alt="Screenshot 2023-05-31 at 00 02 19" src="https://github.com/WFP-VAM/prism-app/assets/3343536/2ae77392-1928-4bdb-9244-901eddb7afd3">

ericboucher commented:
Just confirming that we are ok to move the default title all the way to the left:

BEFORE:
<img width="1728" alt="Screenshot 2023-06-02 at 6 24 59 PM" src="https://github.com/WFP-VAM/prism-app/assets/16843267/ffd86b89-6087-4523-a6f2-b2a16c3f5818">

AFTER:
<img width="1727" alt="Screenshot 2023-06-02 at 6 24 46 PM" src="https://github.com/WFP-VAM/prism-app/assets/16843267/ef4e81a2-1c4b-44c3-b5e8-0545234c5b19">

hav142 commented:
@ericboucher I have updated the grid layout to enhance the responsiveness of the title, particularly for longer titles that were previously displayed incorrectly on tablets or smaller screens. To address this issue, I have moved the default title to the left, which has the same result as the example provided by Amit above, where the title and subtitle were also aligned to the left when they were longer.

@wadhwamatic the translation for the title and subtitle should already be functioning correctly.

wadhwamatic commented:
@hav142 @ericboucher - the previous alignment is preferable. I can do some testing but not for a while. Please try on your side, including with translations.

ericboucher commented:
@hav142 you could simply use a breakpoint `.up('md')` to keep the alignment we had for desktop but remove it for smaller screens. What do you think?

https://mui.com/material-ui/customization/breakpoints/

hav142 commented:
@ericboucher if we want to maintain the previous alignment, we can put back `justifyContent: 'center'` (without breakpoint). Instead, we can set the breakpoint in the Grid component, like `xs={6} md={9}`. This way the logo will have more space on smaller screen, ensuring consistency with larger screen. What do you think? 

![image](https://github.com/WFP-VAM/prism-app/assets/25983260/18bcacc1-0d73-416f-8d40-41445062e7a1)
